### PR TITLE
create block collection

### PR DESCRIPTION
**Pull request by itkg-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1959
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/672
https://github.com/open-orchestra/open-orchestra-model-interface/pull/245
https://github.com/open-orchestra/open-orchestra-elastica-bundle/pull/58
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/314
https://github.com/open-orchestra/open-orchestra-mongo-libs/pull/62

### CODE REVIEW

File: ApiBundle/Controller/AreaController.php [hunk 1]
```diff
@@ -31,220 +32,107 @@ class AreaController extends BaseController
...
-     *
-     * @Config\Route("/{siteId}/{nodeId}/{version}/{language}/move-block", name="open_orchestra_api_area_move_block")
-     * @Config\Method({"POST"})
-     *
-     * @return Response
+     * @return FacadeInterface
      */
-    public function moveBlockFromAreaToAreaAction(Request $request, $nodeId, $language, $version, $siteId)
-    {
+    public function deleteBlockAction(Request $request, $nodeId, $language, $version, $siteId) {
```
Comment (alavieille) [inline ApiBundle/Controller/AreaController.php]: phpdoc

File: ApiBundle/Controller/AreaController.php [hunk 2]
```diff
@@ -31,220 +32,107 @@ class AreaController extends BaseController
...
      * @Config\Method({"POST"})
      *
-     * @return Response
+     * @return FacadeInterface
      */
-    public function moveAreaNodeAction(Request $request, $areaParentId, $nodeId, $language, $version, $siteId)
-    {
-        $areaFacade = $this->get('jms_serializer')->deserialize($request->getContent(), 'OpenOrchestra\ApiBundle\Facade\AreaFacade', $request->get('_format', 'json'));
-
+    public function updateBlockPositionAction(Request $request, $nodeId, $language, $version, $siteId) {
```
Comment (alavieille) [inline ApiBundle/Controller/AreaController.php]: $request is used ?

File: ApiBundle/Transformer/AreaTransformer.php [hunk 3]
```diff
@@ -2,63 +2,50 @@
...
         $this->nodeRepository = $nodeRepository;
-        $this->areaManager = $areaManager;
-        $this->nodeManager = $nodeManager;
     }
 
     /**
      * @param AreaInterface      $area
      * @param NodeInterface|null $node
-     * @param string|null        $parentAreaId
+     * @param string             $areaId
```
Comment (alavieille) [inline ApiBundle/Transformer/AreaTransformer.php]: string|null

File: ApiBundle/Transformer/NodeTransformer.php [hunk 4]
```diff
@@ -168,10 +176,8 @@ protected function addLinks(FacadeInterface $facade, NodeInterface $node)
                 'siteId' => $node->getSiteId()
             )));
 
-           $routeName = 'open_orchestra_api_block_list_without_transverse';
-           if (NodeInterface::TYPE_TRANSVERSE !== $node->getNodeType()) {
-               $routeName = 'open_orchestra_api_block_list_with_transverse';
-            }
+            $routeName = 'open_orchestra_api_block_list_with_transverse';
```
Comment (alavieille) [inline ApiBundle/Transformer/NodeTransformer.php]: this variable is now useless

File: Backoffice/EventSubscriber/WebSiteNodeTemplateSubscriber.php [hunk 5]
```diff
@@ -2,24 +2,24 @@
...
  * Class WebSiteNodeTemplateSubscriber
  */
 class WebSiteNodeTemplateSubscriber implements EventSubscriberInterface
 {
-    protected $templateRepository;
+    protected $templateManager;
 
     /**
-     * @param TemplateRepositoryInterface $templateRepository
+     * @param array $templateSetParameters
```
Comment (alavieille) [inline Backoffice/EventSubscriber/WebSiteNodeTemplateSubscriber.php]: phpdoc TemplateManager

File: Backoffice/EventSubscriber/WebSiteNodeTemplateSubscriber.php [hunk 6]
```diff
@@ -46,15 +59,30 @@ public function onPreSetData(FormEvent $event)
...
+    protected function getTemplateChoices()
+    {
+        $templateSetParameters = $this->templateManager->getTemplateSetParameters();
+        $choices = array();
+        foreach ($templateSetParameters as $keyTemplateSet => $templateSetParameters) {
+            foreach ($templateSetParameters['templates'] as $key => $template) {
+                $choices[$keyTemplateSet][$key] = $template['label'];
+            }
+        }
+        return $choices;
```
Comment (alavieille) [inline Backoffice/EventSubscriber/WebSiteNodeTemplateSubscriber.php]: psr2

File: Backoffice/Manager/NodeManager.php [hunk 7]
```diff
@@ -228,91 +196,46 @@ public function hydrateNodeFromNodeId(NodeInterface $node, $nodeId)
...
-     * @return bool
-     */
-    public function nodeConsistency($nodes)
-    {
-        if (is_array($nodes)) {
-            foreach ($nodes as $node) {
-                if (!$this->areaManager->areaConsistency($node->getRootArea(), $node) || !$this->blockManager->blockConsistency($node)) {
-                    return false;
+            $newNode->setArea($areaId, $newArea);
+            foreach($area->getBlocks() as $block) {
```
Comment (alavieille) [inline Backoffice/Manager/NodeManager.php]: psr2

File: Backoffice/Manager/TemplateManager.php [hunk 8]
```diff
@@ -2,43 +2,120 @@
...
+    }
+
+    /**
+     * @return array
+     */
+    public function getTemplateSetParameters()
+    {
+        return $this->templateSet;
+    }
+    /**
```
Comment (alavieille) [inline Backoffice/Manager/TemplateManager.php]: add empty line

File: ApiBundle/Controller/AreaController.php [hunk 9]
```diff
@@ -31,220 +32,107 @@ class AreaController extends BaseController
...
-    /**
-     * @param AreaFacade $facade
-     * @param string     $nodeId
-     * @param string     $language
-     * @param string     $version
-     * @param string     $siteId
-     */
-    protected function updateArea(AreaFacade $facade, $nodeId, $language, $version, $siteId)
-    {
+            );
```
Comment (ngilain) [inline ApiBundle/Controller/AreaController.php]: bad ident

File: BackofficeBundle/Resources/config/services.yml [hunk 10]
```diff
@@ -80,16 +66,13 @@ services:
...
     open_orchestra_backoffice.manager.keyword_to_document:
         class: '%open_orchestra_backoffice.manager.keyword_to_document.class%'
         arguments:
             - '@open_orchestra_model.repository.keyword'
             - '@open_orchestra_model.helper.suppress_special_character'
             - '%open_orchestra_model.document.keyword.class%'
             - '@security.authorization_checker'
+    open_orchestra_backoffice.manager.template:
+        class: '%open_orchestra_backoffice.manager.template.class%'
+
```
Comment (alavieille) [inline BackofficeBundle/Resources/config/services.yml]: eof

File: BackofficeBundle/Resources/public/less/page/node.less [hunk 11]
```diff
@@ -21,3 +21,12 @@
...
 }
+.area .area-container.area-row .area.block-locked {
+    background: url('../../../images/padlock.png') no-repeat center center;
+    background-color: #333333;
+    h4 {
+        display: block;
+    	color: #ffffff;
+        margin: 0 auto;
+    }
+}
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/less/page/node.less]: eof

File: BackofficeBundle/Resources/public/templateSet/default/default.html [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+            <div class="area">
+                <div class="area-container area-row">
+                    <div class="area block-locked" style="flex: 1 1 0%;">
+                        <h4>footer</h4>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/templateSet/default/default.html]: eof

File: BackofficeBundle/Resources/public/templateSet/full_page.html [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+            <div class="area">
+                <div class="area-container area-row">
+                    <div class="area" style="flex: 1 1 0%;">
+                        <div data-area-id="footer" class="block-open"></div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/templateSet/full_page.html]: eof

File: BackofficeBundle/Resources/public/templateSet/login.html [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+            <div class="area">
+                <div class="area-container area-row">
+                    <div class="area" style="flex: 1 1 0%;">
+                        <div data-area-id="login" class="block-open">login</div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/templateSet/login.html]: eof

File: Backoffice/Manager/NodeManager.php [hunk 15]
```diff
@@ -228,91 +196,46 @@ public function hydrateNodeFromNodeId(NodeInterface $node, $nodeId)
...
-    }
-
-    /**
-     * @param AreaInterface $areaContainer
-     * @param AreaInterface $newAreaContainer
-     */
-    protected function duplicateArea(AreaInterface $areaContainer, AreaInterface $newAreaContainer)
-    {
-        foreach ($areaContainer->getAreas() as $area) {
+        foreach($node->getAreas() as $areaId => $area) {
```
Comment (ngilain) [inline Backoffice/Manager/NodeManager.php]: foreach (

File: Backoffice/Manager/NodeManager.php (hunk 7 above)
Comment (ngilain) [inline Backoffice/Manager/NodeManager.php]: foreach (

File: Backoffice/Manager/TemplateManager.php (hunk 8 above)
Comment (ngilain) [inline Backoffice/Manager/TemplateManager.php]: NL

File: BackofficeBundle/Resources/config/services.yml (hunk 10 above)
Comment (ngilain) [inline BackofficeBundle/Resources/config/services.yml]: NL@EOF

File: BackofficeBundle/Resources/public/less/page/node.less (hunk 11 above)
Comment (ngilain) [inline BackofficeBundle/Resources/public/less/page/node.less]: NL@EOF

File: BackofficeBundle/Resources/public/templateSet/full_page.html (hunk 13 above)
Comment (ngilain) [inline BackofficeBundle/Resources/public/templateSet/full_page.html]: NL@EOF

File: BackofficeBundle/Resources/public/templateSet/login.html (hunk 14 above)
Comment (ngilain) [inline BackofficeBundle/Resources/public/templateSet/login.html]: NL@EOF

**Comment by itkg-nanne** [comment]
update done @ngilain , @alavieille

**Comment by alavieille** [comment]
:+1:

**Pull request merged**